src/AppBundle/Entity/Committee.php

```diff
@@ -0,0 +1,350 @@
+        $this->setName($name);
+        $this->slug = $slug;
+        $this->description = $description;
+        $this->country = $countryCode;
+        $this->postalCode = $postalCode;
+        $this->city = $cityCode;
+        $this->status = $status;
+        $this->approvedAt = $approvedAt;
+        $this->createdAt = $createdAt;
+    }
+
+    public static function createSimple(UuidInterface $uuid, string $creatorUuid, string $name, string $description, string $countryCode): self
+    {
+        return new self(
+            $uuid,
+            Uuid::fromString($creatorUuid),
+            $name,
+            $description,
+            $countryCode
+        );
+    }
+
+    public static function createWithAdherent(Adherent $adherent, string $name, string $description, string $countryCode): self
+    {
+        return new self(
+            self::createUuid($name),
+            clone $adherent->getUuid(),
+            $name,
+            $description,
+            $countryCode
+        );
+    }
+
+    public static function createUuid(string $name)
+    {
+        return Uuid::uuid5(Uuid::NAMESPACE_OID, static::canonicalize($name));
+    }
+
+    public function setLocation(string $postalCode, string $cityCode)
+    {
+        $this->postalCode = $postalCode;
+        $this->city = $cityCode;
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function getSlug(): string
+    {
+        return $this->slug;
+    }
+
+    public function getDescription(): string
+    {
+        return $this->description;
+    }
+
+    public function getCountry(): string
+    {
+        return $this->country;
+    }
+
+    public function getCity()
+    {
+        return $this->city;
+    }
+
+    public function getPostalCode()
+    {
+        return $this->postalCode;
+    }
+
+    public function getFacebookPageUrl()
+    {
+        return $this->facebookPageUrl;
+    }
+
+    public function getTwitterNickname()
+    {
+        return $this->twitterNickname;
+    }
+
+    public function getGooglePlusPageUrl()
+    {
+        return $this->googlePlusPageUrl;
+    }
+
+    public function isWaitingForApproval()
+    {
+        return self::PENDING === $this->status && !$this->approvedAt;
+    }
+
+    public function isApproved()
+    {
+        return self::APPROVED === $this->status && $this->approvedAt;
+    }
+
+    public function isRefused()
+    {
+        return self::REFUSED === $this->status;
+    }
+
+    /**
+     * Marks this committee as approved.
+     *
+     * This method also creates and returns the committee membership
+     * relationship for the adherent who created this committee. The
+     * origin adherent of this committee automatically becomes its
+     * very first host member.
+     *
+     * @param string $timestamp The approval date and time
+     *
+     * @return CommitteeMembership
+     */
+    public function approved(string $timestamp = 'now'): CommitteeMembership
```

Comment:
I think we need to merge this PR first and refactor then with a Worflow component.